api/validation/__init__.py

```diff
@@ -0,0 +1,37 @@
+from functools import wraps
+
+from flask import request, Request
+from marshmallow import Schema, ValidationError
+
+from api.validation.schemas import EC2Action
+
+
+def __valid_request(_request: Request, *, json_schema: Schema = None, args_schema: Schema = None, cookies_schema: Schema = None):
+    errors = {}
+    if json_schema:
+        try:
+            errors.update(json_schema.validate(_request.json))
+        except:
+            raise ValueError('Expected json body')
+
+    if args_schema:
+        errors.update(args_schema.validate(_request.args))
+
+    if cookies_schema:
+        errors.update(cookies_schema.validate(_request.cookies))
+
+    return errors
+
+
+def valid(*, json: Schema = None, args: Schema = None, cookies: Schema = None):
```

Comment:
suggestion(non-blocking): to enhance readability, and since this utility is made to validate requests, I would rename `json` to `body` and `args` to `params`, more related to the HTTP jargon

Reply:
I was actually indecisive between these and the one you propose, so I'll go with yours (which I actually prefer)
Thanks!